font/src/ft/mod.rs

```diff
@@ -529,6 +581,84 @@ impl FreeTypeRasterizer {
     }
 }
 
+fn downsample_bitmap(mut bitmap_glyph: RasterizedGlyph, fixup_factor: f64) -> RasterizedGlyph {
+    // Don't try to upscale
+    if fixup_factor >= 1.0 {
+        return bitmap_glyph;
+    }
+
+    let bitmap_buffer = if let BitmapBuffer::RGBA(bitmap_buffer) = bitmap_glyph.buf {
+        bitmap_buffer
+    } else {
+        return bitmap_glyph;
+    };
+
+    let bitmap_width = bitmap_glyph.width as usize;
+    let bitmap_height = bitmap_glyph.height as usize;
+
+    let width = (bitmap_width as f64 * fixup_factor) as usize;
+    let height = (bitmap_height as f64 * fixup_factor) as usize;
```

Comment:
Calling this `target_width` and `target_height` might make things a lot more clear. Same with `bitmap_width` and `bitmap_height`. Especially since both input and output are bitmaps.